Uniapp cli vue2 运行微信小程序出现构建错误，H5没问题

**问题描述**
使用官方 vue-cli 安装后，启动微信小程序，命令方式与hbuildx方式都出现 Error: Cannot find module 'webpack/lib/GraphHelpers' 问题，应该是 webpack 版本冲突导致的。

之前安装与构建都是成功的，再一次重新安装依赖后出现的；如果说 webpack 导致不行，那么之前应该也不行的。

尝试过使用 yarn、pnpm、npm，node18、node20，这些都尝试过不行，尝试过降级 @vue/cli-service 为 v4 版本，但依赖的很多 eslint、stylelint 等等依赖都得重新处理。

错误信息：

```
Error: Cannot find module 'webpack/lib/GraphHelpers'
Require stack:
- my-project/node_modules/@dcloudio/uni-mp-weixin/lib/independent-plugins/split-independent-chunks-plugin.js
- my-project/node_modules/@dcloudio/uni-mp-weixin/lib/createIndependentPlugin.js
- my-project/node_modules/@dcloudio/vue-cli-plugin-uni/lib/mp/index.js
- my-project/node_modules/@dcloudio/vue-cli-plugin-uni/index.js
- my-project/node_modules/@vue/cli-service/lib/Service.js
- my-project/node_modules/@vue/cli-service/bin/vue-cli-service.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1207:15)
    at Module._resolveFilename (my-project/node_modules/module-alias/index.js:49:29)
    at Module._load (node:internal/modules/cjs/loader:1038:27)
    at Module.require (node:internal/modules/cjs/loader:1289:19)
    at require (node:internal/modules/helpers:182:18)
    at Object.<anonymous> (my-project/node_modules/@dcloudio/uni-mp-weixin/lib/independent-plugins/split-independent-chunks-plugin.js:1:22)
    at Module._compile (node:internal/modules/cjs/loader:1521:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1623:10)
    at Module.load (node:internal/modules/cjs/loader:1266:32)
    at Module._load (node:internal/modules/cjs/loader:1091:12)
```


**复现步骤**

``` shell
# 选择默认模板
vue create -p dcloudio/uni-preset-vue my-project

cd my-project

# H5成功
npm run dev:h5

# 微信小程序失败，错误如上
npm run dev:mp-weixin
```

使用 hHuildX 的菜单启动微信小程序同样会出现错误


**系统信息:**
 - 发行平台: 微信小程序、H5平台
 - 操作系统: Mac 15.6.1 （Arm）
 - HBuilderX: 5.07


**补充信息**

uni-app版本: 
``` text
uni-app v5.07
uni-app cli v2.0.2-5000720260410001

Environment Info:

  System:
    OS: macOS 15.6.1
    CPU: (8) arm64 Apple M1 Pro
  Binaries:
    Node: 20.19.5 - ~/.nvm/versions/node/v20.19.5/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v20.19.5/bin/yarn
    npm: 10.8.2 - ~/.nvm/versions/node/v20.19.5/bin/npm
  Browsers:
    Chrome: 148.0.7778.179
    Firefox: 144.0
    Safari: 18.6
  npmPackages:
    @dcloudio/types: ^3.3.2 => 3.4.31 
    @dcloudio/uni-app: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-app-plus: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-app-plus-nvue:  0.0.1 
    @dcloudio/uni-app-plus-nvue-v8:  0.0.1 
    @dcloudio/uni-automator: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-cli-i18n: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-cli-shared: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-h5: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-helper-json: * => 1.0.13 
    @dcloudio/uni-i18n: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-migration: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-360: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-alipay: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-baidu: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-harmony: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-jd: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-kuaishou: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-lark: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-qq: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-toutiao: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-vue: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-weixin: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-mp-xhs: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-quickapp-native: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-quickapp-webview: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-stacktracey: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-stat: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/uni-template-compiler: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/vue-cli-plugin-hbuilderx: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/vue-cli-plugin-uni: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/vue-cli-plugin-uni-optimize: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/webpack-uni-mp-loader: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @dcloudio/webpack-uni-nvue-loader:  0.0.1 
    @dcloudio/webpack-uni-pages-loader: ^2.0.2-5000720260410001 => 2.0.2-5000720260410001 
    @hap-toolkit/dsl-vue:  0.6.13 
    @vue/babel-helper-vue-jsx-merge-props:  1.4.0 
    @vue/babel-helper-vue-transform-on:  1.5.0 
    @vue/babel-plugin-jsx:  1.5.0 
    @vue/babel-plugin-resolve-type:  1.5.0 
    @vue/babel-plugin-transform-vue-jsx:  1.4.0 
    @vue/babel-preset-app:  5.0.9 
    @vue/babel-preset-jsx:  1.4.0 
    @vue/babel-sugar-composition-api-inject-h:  1.4.0 
    @vue/babel-sugar-composition-api-render-instance:  1.4.0 
    @vue/babel-sugar-functional-vue:  1.4.0 
    @vue/babel-sugar-inject-h:  1.4.0 
    @vue/babel-sugar-v-model:  1.4.0 
    @vue/babel-sugar-v-on:  1.4.0 
    @vue/cli-overlay:  5.0.9 
    @vue/cli-plugin-babel: ~5.0.0 => 5.0.9 
    @vue/cli-plugin-router:  5.0.9 
    @vue/cli-plugin-vuex:  5.0.9 
    @vue/cli-service: ~5.0.0 => 5.0.9 
    @vue/cli-shared-utils:  5.0.9 
    @vue/compiler-core:  3.5.34 
    @vue/compiler-dom:  3.5.34 
    @vue/compiler-sfc:  3.5.34 
    @vue/compiler-ssr:  3.5.34 
    @vue/component-compiler-utils:  3.3.0 (3.3.0)
    @vue/composition-api:  1.7.0 
    @vue/devtools-api:  6.0.0-beta.15 
    @vue/shared: ^3.0.0 => 3.5.34 
    @vue/web-component-wrapper:  1.3.0 
    mpvue-page-factory:  1.0.1 
    mpvue-template-compiler:  1.0.13 
    uni-h5-vue:  2.6.10 
    uni-mp-vue:  2.6.10 
    vue: >= 2.6.14 < 2.7 => 2.6.14 
    vue-hot-reload-api:  2.3.4 
    vue-loader:  17.4.2 (15.11.1, 15.11.1, 17.4.2)
    vue-router:  3.0.1 
    vue-style-loader:  4.1.3 (4.1.3)
    vue-template-compiler: >= 2.6.14 < 2.7 => 2.6.14 (2.6.14)
    vue-template-es2015-compiler:  1.9.1 
    vue3:  1.0.0 
    vuex: ^3.2.0 => 3.6.2 (3.6.2)
  npmGlobalPackages:
    @vue/cli: 5.0.9
```

--- a/packages/uni-cli-shared/lib/style.js
+++ b/packages/uni-cli-shared/lib/style.js
@@ -36,9 +36,9 @@ function createRpx2Unit (unit, unitRatio, unitPrecision) {
    * @param {string | number} val
    * @returns {string}
    */
-  return function(val) {
+  return function (val) {
     if (unitRE.test(val)) {
-      return val.replace(unitRE, function(m, $1) {
+      return val.replace(unitRE, function (m, $1) {
         if (!$1) {
           return m
         }

--- a/packages/uni-mp-weixin/lib/independent-plugins/split-independent-chunks-plugin.js
+++ b/packages/uni-mp-weixin/lib/independent-plugins/split-independent-chunks-plugin.js
@@ -1,4 +1,4 @@
-const GraphHelpers = require('webpack/lib/GraphHelpers');
+const webpack = require('webpack');
 const { normalizePath } = require('@dcloudio/uni-cli-shared');
 const getSplitChunks = require('@dcloudio/vue-cli-plugin-uni/lib/split-chunks');
 const path = require('path');
@@ -136,7 +136,12 @@ class SplitHandler {
             const newChunk = this.compilation.addChunk(chunkName);
             newChunk.chunkReason = 'split chunk for independent';
             for (const module of newChunkInfo.modules) {
-                GraphHelpers.connectChunkAndModule(newChunk, module);
+                if (webpack.version[0] > 4) {
+                   this.compilation.chunkGraph.connectChunkAndModule(newChunk, module);
+                } else {
+                    const GraphHelpers = require('webpack/lib/GraphHelpers');
+                    GraphHelpers.connectChunkAndModule(newChunk, module);
+                }
                 [...newChunkInfo.chunks].forEach(chunk => {
                     if (this.removeModuleFromChunkFilter(chunk)) {
                         chunk.removeModule(module);

